### PR TITLE
Tidy defmt-decoder

### PR DESCRIPTION
**Pull request by Urhengulas**
This PR
* moves the `cfg`-attributes from modules to crate root
* rename `mod logger` to `log`

### CODE REVIEW

File: decoder/src/decoder.rs [hunk 1]
```diff
@@ -4,10 +4,6 @@
 //! not meant to be consumed by other tools at the moment so all the API is unstable.
 
 // NOTE: always runs on the host
```
Comment (jonas-schievink) [inline decoder/src/decoder.rs:6]: This note can also be moved to lib.rs

**Comment by Urhengulas** [comment]
@jonas-schievink What do you think of the dissolving? I hope it reduces a bit of complexity in the imports and file structure.

File: decoder/src/elf2table/mod.rs [hunk 2]
```diff
@@ -12,8 +12,8 @@ use std::{
     path::{Path, PathBuf},
 };
 
-pub use crate::decoder::Table;
-use crate::decoder::{StringEntry, TableEntry, Tag};
+pub use crate::Table;
```
Comment (jonas-schievink) [inline decoder/src/elf2table/mod.rs:15]: This can be made private, there's no longer any reason to reexport it

```suggestion
use crate::Table;
```

**Pull request merged**